app/Providers/AppServiceProvider.php

```diff
@@ -96,6 +96,13 @@ public function register(): void
             $this->app->singleton(\App\Services\TypesenseCollectionDescriptor::class);
         }
         $this->registerModelRepositories();
+
+        // lumen hack for signal handling in artisan commands
+        Signals::resolveAvailabilityUsing(function () {
+            return $this->app->runningInConsole()
+                && ! $this->app->runningUnitTests()
+                && extension_loaded('pcntl');
```

Comment:
@pushrbx Should we add pcntl check when the command runs?

Reply:
Also, should we add a OS:windows check here?

Reply:
This is copy pasta from here:
https://github.com/laravel/framework/blob/9.x/src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php#L211

I trust laravel's code, no more thought needed :smile: